test/test_helper.exs

```diff
@@ -140,9 +140,25 @@ defmodule TestHelper do
       my_char CHAR
     );
 
-    DROP PROCEDURE IF EXISTS single_procedure;
+    # This will only return the trailing ok_packet
+    # because the commands inside the stored procedure
+    # do not return result sets.
```

Comment:
on the wire, do we get an ok_packet for each of the 3 statements or we just get the trailing ok_packet that we get for all procedure calls?

Reply:
For the stored procedure it will only send the trailing ok packet. It seems like only statements that return result sets are sent through the wire when dealing with stored procedures. 

The way I checked this was with these tests:

- https://github.com/elixir-ecto/myxql/pull/152/files#diff-25c00486ed7c3364911a13625fb201869b4221fc96834062c8abe5f099d334daR643
- https://github.com/elixir-ecto/myxql/pull/152/files#diff-25c00486ed7c3364911a13625fb201869b4221fc96834062c8abe5f099d334daR657
- https://github.com/elixir-ecto/myxql/pull/152/files#diff-25c00486ed7c3364911a13625fb201869b4221fc96834062c8abe5f099d334daR671

And then also to make sure the logic for accumulating the results doesn't have an error, I inspected the packets while running the tests.

For the text query I modified the protocol functions like this:

```elixir
def decode_com_query_response(<<0x00, rest::binary>>, "", :initial) do
  IO.inspect "ok packet empty next"
  ok_packet(status_flags: status_flags) = ok_response = decode_ok_packet_body(rest)

  if has_status_flag?(status_flags, :server_more_results_exists) do
    {:cont, {:more_results, ok_response}}
  else
    {:halt, ok_response}
  end
end

def decode_com_query_response(<<0x00, rest::binary>>, _next_data, :initial) do
  IO.inspect "ok packet more data"
  {:cont, {:more_results, decode_ok_packet_body(rest)}}
end

def decode_com_query_response(<<0xFF, rest::binary>>, _next_data, :initial) do
  IO.inspect "error"
  {:halt, decode_err_packet_body(rest)}
end

def decode_com_query_response(payload, next_data, state) do
  IO.inspect "not ok/error"
  decode_resultset(payload, next_data, state, &Values.decode_text_row/2)
end
```
and the only thing that printed was this:

>"ok packet empty next"

And similarly with the prepared statement:

```elixir
def decode_com_stmt_execute_response(<<0x00, rest::binary>>, "", :initial) do
  IO.inspect "execute ok empty next"
  {:halt, decode_ok_packet_body(rest)}
end

def decode_com_stmt_execute_response(<<0xFF, rest::binary>>, "", :initial) do
  IO.inspect "execute error"
  {:halt, decode_err_packet_body(rest)}
end

def decode_com_stmt_execute_response(payload, next_data, state) do
  IO.inspect "execute not ok/error"
  decode_resultset(payload, next_data, state, &Values.decode_binary_row/2)
end
```

the only thing that got printed was this:

> "execute ok empty next"